packages/dev/core/src/Shaders/ShadersInclude/hdrFilteringFunctions.fx

```diff
@@ -186,7 +186,7 @@
         #define inline
         vec3 irradiance(samplerCube inputTexture, vec3 inputN, vec2 filteringInfo
         #ifdef IBL_CDF_FILTERING
-        , sampler2D icdfxSampler, sampler2D icdfySampler
+        , sampler2D icdfxSampler, sampler2D icdfySampler, sampler2D pdfSampler

```

Comment:
I wonder if it's possible to merge two or even all three textures into one... The number of samplers that can be used in a fragment shader is a scarce resource, it would be nice if it were possible to limit this number. This may be something to think about in the future.

Reply:
Unfortunately, `icdfxSampler` and `icdfySampler` are different resolutions.
`pdfSampler` could be packed with `icd` but I was going to explore making it RGB at some point in the future and then using it to improve the colour and intensity of the IBL shadows.

Reply:
I could still combine `icdfxSampler` and `icdfySampler`, I suppose. It would be using more memory, of course, but fewer samplers. And the `pdfSampler` could become the blue channel. If I ever make it RGB, that could be an optional feature.
The memory requirements of the IBL shadows is already a bit concerning.